emcc.py

```diff
@@ -1081,6 +1081,15 @@ def check(input_file):
     if shared.Settings.MODULARIZE_INSTANCE:
       shared.Settings.MODULARIZE = 1
 
+    if shared.Settings.MODULARIZE:
+      assert not options.proxy_to_worker, '-s MODULARIZE=1 and -s MODULARIZE_INSTANCE=1 are not compatible with --proxy-to-worker (if you want to run in a worker with -s MODULARIZE=1, you likely want to do the worker side setup manually)'
+
+    if not shared.Settings.EXPORT_NAME:
+      # If we are building directly to .html with -s MODULARIZE=1 (but not -s MODULARIZE_INSTANCE=1), default to exporting
+      # Emscripten code under function EmscriptenCode() so that it does not collide with the Module object. (When not modularizing,
+      # or if building with MODULARIZE_INSTANCE, or if building to .js manually, there is no conflict, so default to 'Module' in those cases)
+      shared.Settings.EXPORT_NAME = 'EmscriptenCode' if final_suffix == '.html' and shared.Settings.MODULARIZE and not shared.Settings.MODULARIZE_INSTANCE else 'Module'
```

Comment:
This is a breaking change - current users doing `Module()` would no longer work.

How about requiring EXPORT_NAME be set, when MODULARIZE?

Reply:
That works as well - let me update to that.